src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,136 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private MyEntry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new MyEntry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
 
+        if (size + 1 > table.length * LOAD_FACTOR) {
```

Comment:
```suggestion
        if (size >= table.length * LOAD_FACTOR) {
```
put if inside of resize, rename to `resizeIfNeeded` for clarity